lib/spack/spack/test/git_fetch.py

```diff
@@ -160,7 +160,7 @@ def test_git_extra_fetch(tmpdir):
 
     fetcher = GitFetchStrategy(git='file:///not-a-real-git-repo')
     with Stage(fetcher, path=testpath) as stage:
-        mkdirp(stage.source_path)
+        mkdirp(os.path.join(stage.source_path, 'test'))
```

Comment:
Why is this change required? Likewise for `hg_fetch`

Reply:
Because the source_path existing no longer means that the archive was expanded -- the source path could be the same as the stage path, in which case before the archive is expanded exists, but contains nothing but the archive (URL) or nothing (git, svn, hg).

The `expanded` function in the stage was changed for this new behavior, and the tests that are faking an expanded archive had to change to keep up.

Reply:
I think this is no longer the case.